### DISCUSSION TITLE
Let's continue sharing our favorite development tricks

### DISCUSSION THREAD

**Discussion by roblatham00** [roblatham00/development-tricks#1] [Announcements]
## 👋 Welcome!
I'd like to see if "github discussions' might be another way to develop community.   

Of course pull requests are a great way to add content and I'll gladly add anything contributed.

So what might we use discussions for?  Maybe you'd like to workshop an idea.  Maybe you have a problem, and there's no one left to turn to.

**Comment by robertu94** [comment]
Sorry I missed most of the talk.  I was looking forward to, but things ran long.  I looked through the repo, and I have a few suggestions:

https://github.com/roblatham00/development-tricks/blob/0d84923bd4d8ea8cc5d4ca1be3145d62b6c73859/gdb/README.md?plain=1#L10-L20

I've used the SIGSTOP command before.   But, for this specific case I wrote MPIGDB https://github.com/robertu94/mpigdb which writes the syntax for startup for you.  It also adds a few commands to make it easier to work with a set of processes.

https://github.com/roblatham00/development-tricks/blob/0d84923bd4d8ea8cc5d4ca1be3145d62b6c73859/navigation/README.md?plain=1#L5-L6

Now days I just [builtin language server protocol support in NeoVIM](https://github.com/robertu94/dotfiles/blob/b3c4d71e72a7e34941d39d82059852253b0a575a/roles/dotfiles/templates/init.lua.j2#L131-L192), but this also exists for EMACS.  It supports more languages than CSCOPE, but add some other features like show definition on hover, and more.  It's not even insanely hard to setup.  For C/C++/Cuda clangd is the thing you want and it comes with llvm/clang.  For C/C++ you'll need a compile_commands.json.  For CMAKE you can use [`-DCMAKE_EXPORT_COMPILE_COMMANDS=ON`](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html), but if you have something else you can use [bear](https://github.com/rizsotto/Bear)

https://github.com/roblatham00/development-tricks/blob/d2e8beb0caf2929c47fc552473bddd1326fd2ebc/graphviz/README.md?plain=1#L3

GraphViz is great.  I don't use it as much as I could, but I've had some nice results from [d2lang](https://d2lang.com/) which supports some common software dev-y kinds of graphs.

https://github.com/roblatham00/development-tricks/blob/d2e8beb0caf2929c47fc552473bddd1326fd2ebc/shell/README.md?plain=1#L191-L201

I use a wayland based desktop, the equivalent command is `wl-copy` and `wl-paste`

https://github.com/roblatham00/development-tricks/blob/d2e8beb0caf2929c47fc552473bddd1326fd2ebc/learning/README.md?plain=1#L1-L5

I frequently answer questions from students in public on my website.  Here is the index of them.  It includes topics from reading to GDB (including some cool scripting things).  Some of the more developed ones are for software development.

**Reply by roblatham00** [reply]
thanks for the close reading of the supporting materials.   I'm still working on rolling up all the feedback from you and others.  

zoxide is so much fun.
